statistics/selectivity.go

```diff
@@ -269,9 +276,16 @@ func getMaskAndRanges(ctx sessionctx.Context, exprs []expression.Expression, ran
 		accessConds = ranger.ExtractAccessConditionsForColumn(exprs, cols[0].UniqueID)
 		ranges, err = ranger.BuildColumnRange(accessConds, sc, cols[0].RetType, types.UnspecifiedLength)
 	case ranger.IndexRangeType:
-		var res *ranger.DetachRangeResult
-		res, err = ranger.DetachCondAndBuildRangeForIndex(ctx, exprs, cols, lengths)
-		ranges, accessConds, remainedConds, isDNF = res.Ranges, res.AccessConds, res.RemainedConds, res.IsDNFCond
+		if cachedPath == nil {
+			var res *ranger.DetachRangeResult
+			res, err = ranger.DetachCondAndBuildRangeForIndex(ctx, exprs, cols, lengths)
+			ranges, accessConds, remainedConds, isDNF = res.Ranges, res.AccessConds, res.RemainedConds, res.IsDNFCond
```

Comment:
Can we also cache the `mask` and `partCover` instead of passing the `cachedPath` to this function?

Reply:
`mask` and `partCover` is only used in `Selectivity`. So i think we don't need to cache it.